builder/pipeline.cc

```diff
@@ -638,6 +642,11 @@ class pipeline_builder {
       // TODO(vksnk): recomputing this seems really wasteful, we can should be
       // able to maintain the list of buffers as we build the IR.
       symbol_map<bool> buffer_used = buffers_used_inside(body);
+      // Don't really need to emit buffer_crop for base_f, because they will
+      // have crop_dim's anyway.
+      for (const func::output& o : base_f->outputs()) {
+        buffer_used[o.buffer->sym()] = false;
```

Comment:
I think this (and the change below) would be better expressed by adding a check below `if (f == base_f) continue`? I know that right now, buffer_used is only used by that one check, but I think it's a bit tricky.

Reply:
Good point, better now.